### PR TITLE
Add snapshot tests for clickhouse table schemas

### PR DESCRIPTION
**Pull request by macobo**
## Changes

This PR cleans up clickhouse schema definitions and adds tests for both the normal and replicated table definitions. Hoping to use this to make me feel safer about deploying a new table. It also will be of use solving https://github.com/PostHog/posthog/issues/7571

Note that `CLICKHOUSE_REPLICATED` does _not_ work correctly and is out of sync with reality.

## How did you test this code?

Unit tests

### CODE REVIEW

File: ee/clickhouse/sql/test/test_schema.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    PERSONS_DISTINCT_ID_TABLE_MV_SQL,
+    KAFKA_PLUGIN_LOG_ENTRIES_TABLE_SQL,
+    PLUGIN_LOG_ENTRIES_TABLE_SQL,
+    PLUGIN_LOG_ENTRIES_TABLE_MV_SQL,
+    SESSION_RECORDING_EVENTS_TABLE_SQL,
+    SESSION_RECORDING_EVENTS_TABLE_MV_SQL,
+    KAFKA_SESSION_RECORDING_EVENTS_TABLE_SQL,
+]
+
+idfn = lambda query: query if isinstance(query, str) else query()
```
Comment (yakkomajuri) [inline ee/clickhouse/sql/test/test_schema.py:37]: can we give this a better name?

File: ee/clickhouse/sql/clickhouse.py [hunk 2]
```diff
@@ -46,23 +40,25 @@
...
 def kafka_engine(
     topic: str,
-    kafka_host=KAFKA_HOSTS,
+    kafka_host=None,
     group="group1",
     serialization="JSONEachRow",
     proto_schema=None,
     skip_broken_messages=100,
 ):
+    if kafka_host is None:
```
Comment (yakkomajuri) [inline ee/clickhouse/sql/clickhouse.py:60]: legitimately curious: why?

**Review by yakkomajuri — APPROVED** [review]
lgtm

File: ee/clickhouse/sql/clickhouse.py (hunk 2 above)
Comment (macobo) [reply ee/clickhouse/sql/clickhouse.py:60]: So we'd be able to override it in tests. Premature optimization from me, didn't need to be here.

**Pull request merged**